src/main/java/com/wanderersoftherift/wotr/gui/tooltip/GearImplicitsTooltipEvent.java

```diff
@@ -35,21 +33,17 @@ public static void on(RenderTooltipEvent.GatherComponents event) {
         if (implicits == null) return;
         List<ModifierInstance> modifierInstances = implicits.modifierInstances(stack, Minecraft.getInstance().level);
 
-        List<TooltipComponent> toAdd = new ArrayList<>();
+        list.add(1, Either.left(Component.translatable("tooltip."+ WanderersOfTheRift.MODID +".implicit").withStyle(ChatFormatting.GRAY)));
 
+        List<TooltipComponent> toAdd = new ArrayList<>();
         for (ModifierInstance modifierInstance : modifierInstances) {
-
-                float roll = modifierInstance.roll();
-                float roundedValue = (float) (Math.ceil(roll * 100) / 100);
-
-                // TODO: Hardcoded currently, need to see how the modifier stuff develops further
-                MutableComponent cmp = Component.literal("+" + roundedValue + " " + modifierInstance.modifier().getRegisteredName()).withStyle(ChatFormatting.AQUA);
-                toAdd.addLast(new ImageTooltipRenderer.ImageComponent(stack, cmp, WanderersOfTheRift.id("textures/tooltip/attribute/damage_attribute.png")));
+            List<TooltipComponent> tooltipComponents = modifierInstance.getTooltipComponent(stack, ChatFormatting.AQUA);
+            toAdd.addAll(tooltipComponents);
         }
 
 
         for (int i = 0; i < toAdd.size(); i++) {
-            list.add(i + 1, Either.right(toAdd.get(i)));
+            list.add(i + 2, Either.right(toAdd.get(i)));
```

Comment:
I wonder if something like

```
list.addAll(2, toAdd.stream().map(Either::right).toList());
```

would work here.

Reply:
Probably, I do have some concerns on order, but I think that should generally be okay. Will probably also count for the Sockets.

Reply:
After consideration, we should really combine the tooltips into one class. a1qs had some remarks on it as well.